libs/libarchfpga/src/physical_types.h

```diff
@@ -62,6 +65,74 @@ struct t_cluster_placement_primitive;
 struct t_arch;
 enum class e_sb_type;
 
+/****************************************************************************/
+/* FPGA metadata types                                                      */
+/****************************************************************************/
+/* t_offset, t_metadata_as, and t_metadata_dict provide a types to store
+ * metadata about the FPGA architecture and routing routing graph along side
+ * the pb_type, grid, node and edge descriptions.
+ *
+ * The metadata is stored as a simple key/value map.  key's are string and an
+ * optional coordinate. t_metadata_as provides the value storage, which is a
+ * string.  t_metadata_as has some convience methods for a handful of types
+ * (e.g. int, vector of int), but the underlying data type is always a string.
+ */
+
+// Metadata value storage.
+class t_metadata_as {
+public:
+    explicit t_metadata_as(std::string v) : value_(v) {}
+    explicit t_metadata_as(const t_metadata_as &o) : value_(o.value_) {}
+
+    // Return string value.
```

Comment:
I think a better name for this class would be t_metadata_val or t_metadata_value instead of t_metadata_as.  The class stores the value, and has an as_string method, but it isn't a type conversion class so its current name is counter-intuitive to me.